src/hotspot/share/runtime/thread.hpp

```diff
@@ -639,6 +639,11 @@ class Thread: public ThreadShadow {
 
  private:
   VMErrorCallback* _vm_error_callbacks;
+
+  bool  _profile_vm_locks;
```

Comment:
It is not clear what role this plays - why do we only profile certain threads?

Reply:
For now, we only profile the main thread.
It is used for determining whether to start and stop the "wait time" and "hold time" timers. (Please refer to changes in mutexLocker.hpp)